federatedscope/attack/worker_as_attacker/server_attacker.py

```diff
@@ -164,14 +164,14 @@ def __init__(self,
             fl_model_criterion=get_criterion(self._cfg.criterion.type,
                                              device=self.device),
             device=self.device,
-            grad_clip=self._cfg.optimizer.grad_clip,
+            grad_clip=self._cfg.grad.grad_clip,
             dataset_name=self._cfg.data.type,
             fl_local_update_num=self._cfg.federate.local_update_steps,
-            fl_type_optimizer=self._cfg.fedopt.type_optimizer,
+            fl_type_optimizer=self._cfg.fedopt.optimizer.type,
```

Comment:
Should we ensure forward compatibility? That is, both optimizer.grad_clip or grad.grad_clip are ok. Otherwise, we may have to exhaustively modify the historical codes to ensure that the previous experiments still work correctly

Reply:
- @yxdyc Thanks, I have checked the historical codes and make sure that all unittest works well. 
- As for gradient clipping, since it is not a common parameter in general optimizers (like the learning rate), maybe we should consider it as an independent operation and separate it from the optimizer. 
- Since we have set `cfg.optimizer = CN(new_allowed=True)`, our modification also supports `optimizer.grad_clip` as a parameter for customized optimziers.